### ISSUE TITLE
Allow CharacterizeJob to receive either an AF object or a Valkyrie resource as the FileSet

### ISSUE THREAD

**Issue by elrayle** [samvera/hyrax#4100]
### Descriptive summary

Convert CharacterizeJob to allow it to receive either an ActiveFedora based FileSet or a Valkyrie::Resource equivalent FileSet.

### Rationale

Allow jobs to execute correctly with existing AF implementation or the wings Valkyrie implementation.

### Expected behavior

CharacterizeJob receives a Valkyrie::Resource and performs the characterization.

### Actual behavior

CharacterizeJob raises an exception when passed a Valkyrie::Resource.

### Approach for Conversion

See [Pattern for Jobs Receiving Parameter that may be an AF Object or a Valkyrie Resource](https://docs.google.com/document/d/1f9yDjtkXapFf1DYLyPhprnB4CPGaZw9p_k8T_tcor-U/edit#heading=h.qt9rxydfd727)

### Steps to reproduce the behavior

1. Need to add tests that validate that the same tests pass whether a Valkyrie::Resource or an AF object is passed.

See example of processing both objects in...  [file_set_actor_spec.rb](https://github.com/samvera/hyrax/blob/master/spec/actors/hyrax/actors/file_set_actor_spec.rb#L6)

For testing this job, you can use something like...
```
RSpec.describe CharacterizeJob do
  [true, false].each do |test_valkyrie|
    context "when test_valkyrie is #{test_valkyrie}" do
      let(:af_file_set) { create(:file_set) }
      let(:valk_file_set) { af_file_set.valkyrie_resource }
      let(:file_set) { test_valkyrie ? valk_file_set : af_file_set }
```

OR you can write tests that are specific to each object type, similar to the tests in [spec/jobs/hyrax/revoke_edit_job_spec.rb](https://github.com/samvera/hyrax/blob/master/spec/jobs/hyrax/revoke_edit_job_spec.rb).  This approach doubles the number of tests and doesn't provide certainty that the new code works exactly the same as the old code if the two versions of the tests diverge.  The approach to take is at the discretion and judgement of the developer.

### Related work

These have all be modified to support AF object or Valkyrie resource using the [established pattern]((https://docs.google.com/document/d/1f9yDjtkXapFf1DYLyPhprnB4CPGaZw9p_k8T_tcor-U/edit#heading=h.qt9rxydfd727). 

* Issue #4101 Allow GrantEditToMembersJob to receive either an AF object or a Valkyrie resource as the Work (PR #4123)
* Issue #4102 Allow GrantReadToMembersJob to receive either an AF object or a Valkyrie resource as the Work (PR #4123)
* Issue #4103 Allow RevokeReadToMembersJob to receive either an AF object or a Valkyrie resource as the Work (PR #4123)
* Issue #4099 Add GlobalID::Identification for resources passed to jobs (PR #4124)

**Comment by elrayle** [comment]
Fixed by PR #4123

**Comment by mjgiarlo** [comment]
Blocked by #4060. Should be unblocked by #4055.

**Comment by no-reply** [comment]
I think this might be a wontfix, since there's now a characterization infrastructure that works with Valkyrie and is used for Valkyrie driven objects by default

**Comment by hackartisan** [comment]
This work was done in https://github.com/samvera/hyrax/pull/5321
